docs/data/material/migration/migrating-to-v6/migrating-to-pigment-css.md

```diff
@@ -0,0 +1,484 @@
+      }}
+      style={{
+        '--offset': `${index}px`,
+      }}
+    />
+  ))}
+</div>
+```
+
+</codeblock>
+
+### styled
+
+If you have a custom components that are using `styled` from `@mui/material/styles`, change the import source to `@mui/material-pigment-css`:
+
+```diff
+- import { styled } from '@mui/material/styles';
++ import { styled } from '@mui/material-pigment-css';
+```
+
+Then, run the following codemod:
+
+```bash
+npx @mui/codemod@next v6.0.0/styled path/to/folder
+```
+
+The scenarios below are not covered by the codemod, you have to manually update them:
+
+#### Dynamic styles based on props
+
+If you have dynamic styles based on props, you need to move them to CSS variables. You will need to create a wrapper component to set the inline style with the CSS variables.
+
+<codeblock>
+
+```js before
+const FlashCode = styled('div')(
+  ({ theme, startLine = 0, endLine = startLine, lineHeight = '0.75rem' }) => ({
+    top: `calc(${lineHeight} * 1.5 * ${startLine})`,
+    height: `calc(${lineHeight} * 1.5 * ${endLine - startLine + 1})`,
+    ...theme.typography.caption,
+  }),
+);
+
+export default FlashCode;
+```
+
+```js after
+const FlashCodeRoot = styled('div')(({ theme }) => ({
+  top: `calc(var(--Flashcode-lineHeight) * 1.5 * var(--Flashcode-startLine))`,
+  height: `calc(var(--Flashcode-lineHeight) * 1.5 * (var(--Flashcode-endLine) - var(--Flashcode-startLine) + 1))`,
+  ...theme.typography.caption,
+}));
+
+const FlashCode = React.forwardRef(function FlashCode(props, ref) {
+  const {
+    children,
+    startLine = 0,
+    endLine = startLine,
+    lineHeight = '0.75rem',
+    ...other
+  } = props;
+
+  return (
+    <FlashCodeRoot
+      ref={ref}
+      {...other}
+      style={{
+        '--Flashcode-lineHeight': lineHeight,
+        '--Flashcode-startLine': startLine,
+        '--Flashcode-endLine': endLine,
+        ...other.style,
+      }}
+    >
+      {children}
+    </FlashCodeRoot>
+  );
+});
+
+export default FlashCode;
+```
+
+</codeblock>
+
+## Migrate layout components
+
+Material UI v6 provides a separate set of layout components, including `Grid`, `Container`, and `Stack`, that are compatible with Pigment CSS.
+
+```diff
+- import Container from '@mui/material/Container';
++ import Container from '@mui/material/PigmentContainer';
+
+- import Grid from '@mui/material/Grid';
++ import Grid from '@mui/material/PigmentGrid';
+
+- import Stack from '@mui/material/Stack';
++ import Stack from '@mui/material/PigmentStack';
+
+- import Hidden from '@mui/material/Hidden';
++ import Hidden from '@mui/material/PigmentHidden';
+```
+
+For Box component, you can remove the import and use the HTML element directly.
+
+```diff
+- import Box from '@mui/material/Box';
+
+ function CustomCard() {
+   return (
+-    <Box sx={{ display: 'flex' }}>
+-      <Box component="img" src="..." sx={{ width: 24, height: 24 }}>
+-      ...
+-    </Box>
++    <div style={{ display: 'flex' }}>
```

Comment:
Did you mean here to use `style` with the `div` or should it be `sx`?
- based on the TypeScript type extension of the global for HTML attributes, I believe you meant to specify `sx` here

Reply:
Opps, that's a mistake.